documentation/decisions/adr/2023-12-18-measurement-dashboard-architecture.md

```diff
@@ -0,0 +1,149 @@
+
+There are a number of architectural patterns we can follow to build and publish a set of dashboards, and the pattern we choose will influence how we set up our data engineering pipeline, as well as our infrastructure for analytics and business intelligence (ABI).
+
+The goal of this ADR is to evaluate and recommend an architectural approach for our initial public measurement dashboard deliverable, which will focus on sprint and delivery metrics. However, this recommendation should also create a flexible yet robust foundation for future dashboards as well.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+#### Must have
+
+- Members of the public can access the dashboard without a login
+- Multiple related dashboards can be combined into a single "application" for external users
+- New charts and dashboards are easy to implement
+- Users can export or download the data behind a given dashboard
+- Open source contributors can host a local version of our dashboards using exported data
+
+#### Nice to have
+
+- The platform used to build dashboards can also be used to answer ad-hoc questions about the data
+- System-to-system users can programmatically access the data behind a given dashboard via API
+- Dashboards can include narrative text that explains the metrics we publish
+
+## Options Considered
+
+- Analytics API + user interface
+- Custom dashboard application (e.g. [Dash][plotly-dash] or [R Shiny][r-shiny])
+- Open source dashboard solution (e.g. [Metabase][metabase] or [Redash][redash])
+- SaaS dashboard solution (e.g. [PowerBI][power-bi] or [Tableau][tableau])
+
+## Decision Outcome <!-- REQUIRED -->
+
+TODO: Decide between analytics API + UI and open source dashboard solution.
+
+### Positive Consequences <!-- OPTIONAL -->
+
+- {e.g., improved performance on quality metric, new capability enabled, ...}
+- ...
+
+### Negative Consequences <!-- OPTIONAL -->
+
+- {e.g., decreased performance on quality metric, risk, follow-up decisions required, ...}
+- ...
+
+## Pros and Cons of the Options <!-- OPTIONAL -->
+
+### Analytics API + user interface
+
+This option involves building both a custom analytics API that will serve the data behind our key project metrics and a separate user interface to consume from that API and render these metrics as a dashboard for end users. Following this approach, we could choose to re-use our existing front-end and backend infrastructure or make slightly different tooling choices based on our needs.
+
+> [!TIP]
+> **Bottom line:** This option is best if we want to follow the existing architecture pattern of Simpler.Grants.gov and have maximum control over the structure and design of the public dashboard, but can dedicate more engineering resources to building and hosting this dashboard and don't need to update the analytics endpoints or dashboards frequently.
```

Comment:
Yeah, this feels like the "slow and steady" option. It's easy to understand, hard and slow to update, and takes a lot of work. Not very agile but very reliable.